server/src/main/java/org/opensearch/wlm/Rule.java

```diff
@@ -0,0 +1,370 @@
+            in.readString(),
+            Feature.fromName(in.readString())
+        );
+    }
+
+    public static void requireNonNullOrEmpty(String value, String message) {
+        if (value == null || value.isEmpty()) {
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    public static void validatedAttributeMap(Map<RuleAttribute, Set<String>> attributeMap, Feature feature) {
+        if (!featureAlloedAttributesMap.containsKey(feature)) {
+            throw new IllegalArgumentException("Couldn't find any valid attribute name under the feature: " + feature.getName());
+        }
+        Set<RuleAttribute> ValidAttributesForFeature = featureAlloedAttributesMap.get(feature);
+        for (Map.Entry<RuleAttribute, Set<String>> entry : attributeMap.entrySet()) {
+            RuleAttribute ruleAttribute = entry.getKey();
+            Set<String> attributeValues = entry.getValue();
+            if (!ValidAttributesForFeature.contains(ruleAttribute)) {
+                throw new IllegalArgumentException(
+                    ruleAttribute.getName() + " is not a valid attribute name under the feature: " + feature.getName()
+                );
+            }
+            if (attributeValues.size() > 10) {
+                throw new IllegalArgumentException(
+                    "Each attribute can only have a maximum of 10 values. The input attribute " + ruleAttribute + " exceeds this limit."
+                );
+            }
+            for (String attributeValue : attributeValues) {
+                if (attributeValue.isEmpty()) {
+                    throw new IllegalArgumentException("Attribute value should not be an empty string");
+                }
+                if (attributeValue.length() > 100) {
+                    throw new IllegalArgumentException(
+                        "Attribute value can only have a maximum of 100 characters. The input " + attributeValue + " exceeds this limit."
+                    );
+                }
+            }
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(_id);
+        out.writeMap(attributeMap, RuleAttribute::writeTo, StreamOutput::writeStringCollection);
+        out.writeString(label);
+        out.writeString(updatedAt);
+        out.writeString(feature.getName());
+    }
+
+    public static Rule fromXContent(final XContentParser parser) throws IOException {
+        return Builder.fromXContent(parser).build();
+    }
+
+    public String get_id() {
+        return _id;
+    }
+
+    public String getLabel() {
+        return label;
+    }
+
+    public String getUpdatedAt() {
+        return updatedAt;
+    }
+
+    public Feature getFeature() {
+        return feature;
+    }
+
+    public Map<RuleAttribute, Set<String>> getAttributeMap() {
+        return attributeMap;
+    }
+
+    /**
+     * This Feature enum contains the different feature names for each rule.
```

Comment:
We should put it differently, something like ** ... enum enumerates the features that can use the Rule Based Auto-tagging**